torch/onnx/_internal/exporter/_ir_passes.py

```diff
@@ -30,6 +31,53 @@ def rename_outputs(model: ir.Model, new_names: Sequence[str]) -> None:
         output.name = new_name
 
 
+def _all_values(model: ir.Model):
+    """Yield all values in a model."""
+    # Yield all values in the model
+    yield from model.graph.inputs
+    yield from model.graph.initializers.values()
+    for node in ir.traversal.RecursiveGraphIterator(model.graph):
+        yield from node.outputs
+    # Yield all values in functions
+    for function in model.functions.values():
+        yield from function.inputs
+        for node in ir.traversal.RecursiveGraphIterator(function):
+            yield from node.outputs
+
+
+def _replace_names(shape_expr: str, rename_mapping: dict[str, str]) -> str:
+    """Replace all known names in a shape expression with new names."""
+    for old_name, new_name in rename_mapping.items():
+        shape_expr = re.sub(rf"\b{old_name}\b", new_name, shape_expr)
+    return shape_expr
+
+
+def rename_axis(model: ir.Model, rename_mapping: dict[str, str]) -> None:
```

Comment:
I would check a few cases when the mapping contains strings that is not in the model; I would also have a test the ensure that shapes in functions are renamed.

Reply:
Done. I need to add a sorting step to mapping because there could be the multiple matching cases, and we want the longest string match, for example, 

{"s1": "seq_len", "s2": "past_seq_len", "s1 + s2": "masked_seq_len"}

This is fine when we check the key and match the value, but it fails when we use it on re.sub. LMK if this change makes sense.

Reply:
Sorting works I guess!